samples/openapi3/client/petstore/typescript/builds/enum/models/Interaction.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Sample API to test enum mapping
+ * API description in Markdown.
+ *
+ * OpenAPI spec version: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import { ChatMessage } from '../models/ChatMessage.ts';
+import { Email } from '../models/Email.ts';
+import { HttpFile } from '../http/http.ts';
+
+export class Interaction {
+    /**
+    * The type of interaction
+    */
+    'type': InteractionTypeEnum;
+    /**
+    * The message content
+    */
+    'message'?: string;
+    /**
+    * The email subject
+    */
+    'subject'?: string;
+    /**
+    * The email body
+    */
+    'body'?: string;
+
+    static readonly discriminator: string | undefined = "type";
+
+    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
+        {
+            "name": "type",
+            "baseName": "type",
+            "type": "InteractionTypeEnum",
+            "format": ""
+        },
+        {
+            "name": "message",
+            "baseName": "message",
+            "type": "string",
+            "format": ""
+        },
+        {
+            "name": "subject",
+            "baseName": "subject",
+            "type": "string",
+            "format": ""
+        },
+        {
+            "name": "body",
+            "baseName": "body",
+            "type": "string",
+            "format": ""
+        }    ];
+
+    static getAttributeTypeMap() {
+        return Interaction.attributeTypeMap;
+    }
+
+    public constructor() {
+    }
```

Comment:
This is the meaningful part of the fixture (the omission of `this.type = "Interaction"`)